pkg/aggregator/demultiplexer_agent.go

```diff
@@ -433,31 +434,17 @@ func (d *AgentDemultiplexer) flushToSerializer(start time.Time, waitForSerialize
 			}
 		}, func(serieSource metrics.SerieSource) {
 			sendIterableSeries(d.sharedSerializer, start, serieSource)
+		},
+		func(sketches metrics.SketchesSource) {
+			if sketches.WaitForValue() {
```

Comment:
Why do we need to wait for a value here, but not in `sendIterableSeries`?

Reply:
Unlike sketches, there are always series to send (For example `datadog.agent.running`)

Reply:
Letting the serializer handles this case was an option too. I decided to use `WaitForValue()` because there are two serializers for sketches.